Tasks/AzureFileCopy/Utility.ps1

```diff
@@ -485,30 +493,36 @@ function Get-MachinesFqdnsForPublicIP
         {
             if(-not [string]::IsNullOrEmpty($publicIP.DnsSettings.Fqdn))
             {
-                $fqdnMap[$publicIp.IpConfiguration.Id] =  $publicIP.DnsSettings.Fqdn
+                $fqdnMap[$publicIp.IpConfiguration.Id.ToLower()] =  $publicIP.DnsSettings.Fqdn
```

Comment:
While I understand the need for tolower() for VMid, I am not sure about rest of resources. Ensure to test that we are not breaking for rest of the resources.